src/front/app/components/globals/Input.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+const classNames = require('classnames');
+
+const inputTypes = [
+  // Input Based
+  'text',
+  'email',
+  'password',
+  'number',
+  'date',
+  'datetime',
+  'datetime-local',
+  'search',
+  'tel',
+  'color',
+  'file',
+  // Select Based
+  'select',
+  // Text Area Based
+  'textarea',
+];
+
+class Input extends Component {
+  static propTypes = {
+    inputType: PropTypes.oneOf(inputTypes).isRequired,
+    name: PropTypes.string.isRequired,
+    id: PropTypes.string,
```

Comment:
This is required. Don't make it optional

Reply:
id is not required within any **type** of input. Only **name** is required for an input to work properly within any sort of form, as far as I'm aware.

Reply:
If you're moving ahead to doing Labels & FormGroups, Labels do require an ID to work.

Reply:
Labels don't require ID either. Labels require the **name** , which is used within the **for** attribute of labels.

Reply:
No, they use ID: https://www.w3schools.com/tags/tag_label.asp